src/home/homed-home.h

```diff
@@ -0,0 +1,168 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+typedef struct Home Home;
+
+#include "homed-manager.h"
+#include "homed-operation.h"
+#include "list.h"
+#include "ordered-set.h"
+#include "user-record.h"
+
+typedef enum HomeState {
+        HOME_UNFIXATED,               /* home exists, but local record does not */
+        HOME_ABSENT,                  /* local record exists, but home does not */
+        HOME_INACTIVE,                /* record and home exist, but is not logged in */
+        HOME_FIXATING,                /* generating local record from home */
+        HOME_FIXATING_FOR_ACTIVATION, /* fixating in order to activate soon */
+        HOME_FIXATING_FOR_ACQUIRE,    /* fixating because Acquire() was called */
```

Comment:
Those names too are very cryptic. Fixating means either "pay neurotic attention" or "make fixed or stationary", and neither of those meanings reflect what is being done here. E.g. HOME_CREATING_STORAGE would go a long way towards making the code easier to follow.

Reply:
hmm, but that's not what it is. "fixating" is what i call the operation when homed found some home directory somewhere (maybe you plugged in a USB stick or so) and you actually decide to log into that. At that moment we need to make the UID assignment permanent, i.e. need to store everything we know about the USB medium to the host, so that from that point on the user record is "fixated", i.e. means one and one thing only: has a fixed UID assigned and refers to a very specific USB medium and so on. Fixating hence just means: take whatever we already know about the home directory, check if it all makes sense, and write it to an identity record in /var/lib/home/.

or in other words: there are "volatile" and "fixated" user records: a "volatile" record appears because you plug some random usb stick into your machine, but that is just a construct in memory. As soon as you fixate it (which you do at login, but can also request explicitly) it becomes "fixated" and thus becomes permanent with a matching file in /var/lib/home/*.identity.

And yupp I think "make fixed or stationary" is exactly what this is, no?

Reply:
maybe "pinning"? "fixating" nowadays really is almost exclusively used in the sense of "focus attention on" (or [in this sense](https://en.wikipedia.org/wiki/Carbon_fixation) which also isn't really what you mean).

Reply:
if the idea is opposite of volatile, I think of static, stable, immutable; in this case maybe ready, settled, preparing or prepared. Fixate does have either a vision or obsession connotation. Fixed is better, but could be confused with fsck repair, journal replay, or (btrfs) self-healing.